optimum/exporters/openvino/__main__.py

```diff
@@ -31,13 +37,113 @@
 from .convert import export_models
 
 
+if TYPE_CHECKING:
+    from transformers import PreTrainedModel, TFPreTrainedModel
+
+
 OV_XML_FILE_NAME = "openvino_model.xml"
 
 _MAX_UNCOMPRESSED_SIZE = 1e9
 
 logger = logging.getLogger(__name__)
 
 
+def _get_submodels_and_export_configs(
+    model: Union["PreTrainedModel", "TFPreTrainedModel"],
+    task: str,
+    custom_onnx_configs: Dict,
+    custom_architecture: bool,
+    _variant: str,
+    int_dtype: str = "int64",
+    float_dtype: str = "fp32",
+    fn_get_submodels: Optional[Callable] = None,
+    preprocessors: Optional[List[Any]] = None,
+    no_position_ids: bool = False,
+):
+    is_stable_diffusion = "stable-diffusion" in task
+    if not custom_architecture:
+        if is_stable_diffusion:
+            onnx_config = None
+            models_and_onnx_configs = get_stable_diffusion_models_for_export(
+                model, int_dtype=int_dtype, float_dtype=float_dtype
+            )
+        else:
+            onnx_config_constructor = TasksManager.get_exporter_config_constructor(
+                model=model, exporter="openvino", task=task
```

Comment:
here I would prefer we keep the onnx config so that we don't duplicate code as I'm unsure why we would need it for now, is there a specific reason for that @eaidova ?

Reply:
the main idea of this pr to allow override configs for onnx for openvino.
I see 2 reasons for that:
1. Adding configurations for supporting new models now requires to guiranree that model can be exported in onnx (and possibly even provide pipeline for running it with ORT). However, we do not use export to onnx as default path for export models to openvino anymore. The set of supported operations from torch in openvino can be different from set of ops supported to torch-onnx export. For example, we successfully convert Tapas model family to openvino (for example https://huggingface.co/google/tapas-tiny-finetuned-wtq), but onnx export failed with unsupported aten::scatter_reduce
2. Unblocking different optimizations of existing models that specific for openvino. For the same reasons that we do not export models to onnx and has own inference pipelines, in some time it may happens that onnx and openvino export paths can become different and requires different configurations (for example if I want to merge 2 decoders in seq2seq models in way how it will be convenient for openvino or in some other our plans we want to try export caulsallm models with including beam search inside models). Some simple example, where onnx configuration is not perfectly fit for us, text-generation-with-past, if I right understand, for onnx is default path is merged model, while we use model-with-past. ONNX configs for model with past fill input_ids with only 1 token for this case that lead to some exporting models troubles, that requires models patching and consider each new case as separated (in the latest optimum, created patchers per model, now there is just function that check model type and apply patch on pytorch model), but it can be avoided in majority cases if input_ids will have 2 tokens instead of 1.  Also it uses dynamic only batch for this case, that lead to extra model reshaping before loading to make sequence len dynamic for this input.
3. Simplification of new models enabling flow. We really like optimum and all its features for its smooth user experience (thank you very much for everything what you do) and recommend it to our openvino users as the main path for running inference for HF models using openvino, so we are very interested in extension of supported models and having the latest trading models available running with openvino. Support of cli and API for export models open the door for converting everything that supported in optimum directly in openvino (even if we do not have some OVModelForXXX classes) directly to openvino and it is the great step for us.  But now, it is not enough to just install optimum-intel from git for get the latest available models from optimum side, it requires also install optimum for git. But there is no guarantee that they are synchronized. Like you already highlighted about changes with position_ids for example or another thing what I recently found trying to run mistral model that now for some models it is not enough to specify only with_past=True for getting model with past in inputs and outputs in the same time, additional with_past_in_inputs flag should be passed in config. We need to wait the next official realse for aligning and getting new models supported that maybe non-convinient for us.

We do not duplicate code, (for majority cases configs mapping filled in runtime, just reusing the same onnx config, but if we have own one for openvino specifc, we will use own) just allow to override some export configs if it is applicable.